samples/browser/src/main/java/org/mozilla/samples/browser/SampleApplication.kt

```diff
@@ -59,7 +59,10 @@ class SampleApplication : Application() {
                         val selected = components.sessionManager.findSessionById(sessionId)
                         selected?.let { components.tabsUseCases.selectTab(it) }
                 },
-                onUpdatePermissionRequest = components.addonUpdater::onUpdatePermissionRequest
+                onUpdatePermissionRequest = components.addonUpdater::onUpdatePermissionRequest,
+                onInstallExtensions = {
+                    components.addonUpdater.registerAllInstalledAddonsForFutureUpdates()
```

Comment:
Here you're passing the list of extensions to the `onInstalledExtensions` lambda. Why not simply forward that list (pass along as a parameter) to `registerAllInstalledAddonsForFutureUpdates`, instead of reading it again from the map?

Reply:
Agree 👍 
I did  to keep the name consistency `registerAllInstalledAddonsForFutureUpdates` (**AllInstalledAddons**) as if It was taking a list of extensions the name could be a bit ambiguous, as it wouldn't be doing what is name says.  **But** I re-wrote it in more concise way, just overloading `registerForFutureUpdates` to accept a list of `webExtensions`